cmd/airflow.go

```diff
@@ -500,7 +502,16 @@ func airflowStart(cmd *cobra.Command, args []string) error {
 	if len(args) > 0 {
 		envFile = args[0]
 	}
-
+	// check if ports are allocated
+	err := checkPort(config.CFG.PostgresPort.GetString())
+	if err != nil {
+		return errors.Wrap(err, "the Postgres port "+config.CFG.PostgresPort.GetString()+" is already in use. Either Airflow is already running in this Astro project, another Astro project, or another service is using this port")
```

Comment:
IMO if the project is already running, then the `cannot start, project already running` error should take precedence over this error, so ideally both these checks should be in the `airflow` package after `d.composeService.Ps` in the `Start` method

Reply:
d.composeService.Ps doesn't run till after the image is built. This can be very ignoring for users who's project takes 5+min to build. They have to wait all that time just to learn that the ports are already allocated

Reply:
`d.composeService.Ps` check runs upfront, before building any images or any other operation.

Reference: https://github.com/astronomer/astro-cli/blob/main/airflow/docker.go#L159-L172

<img width="1282" alt="Screenshot 2022-11-22 at 10 12 42 AM" src="https://user-images.githubusercontent.com/92356010/203223519-27029c7f-dce7-4485-b30c-a67532c06c64.png">